### PR TITLE
fix: support OpenAPI 3.1 type arrays

### LINKED ISSUE #29: [Bug] json: cannot unmarshal array into Go struct field TBis.components.schemas.properties.type of type string
1. start with a swagger openapi v3.1.0 json file
2. exec --input attendance.json  --output attendance-mcp.yaml --server-name attendance
3. error is Error parsing OpenAPI specification: failed to parse OpenAPI document: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal array into Go struct field TBis.components.schemas.properties.type of type string

go version go1.26.1 linux/amd64

### PR DESCRIPTION
**Pull request by Aias00**
## Summary
- upgrade kin-openapi to v0.124.0 so OpenAPI 3.1 schemas with array-valued `type` parse correctly
- normalize schema type handling in the converter so nullable and multi-type schemas still map to MCP args reliably
- add parser and converter regression tests for `type: ["string", "null"]`

## Testing
- go test ./...

Fixes #29

### CODE REVIEW

File: pkg/converter/converter_test.go [hunk 1]
```diff
@@ -11,9 +11,66 @@ import (
...
+						"required": true,
+						"content": {
+							"application/json": {
+								"schema": {
+									"type": "object",
+									"required": ["nickname"],
+									"properties": {
+										"nickname": {
+											"type": ["string", "null"],
+											"description": "Optional nickname"
```
[the remark this answers is not in the document]
Comment (Aias00) [reply pkg/converter/converter_test.go]: Updated the fixture description to "Required nickname (nullable string)" so the test now matches the required-but-nullable semantics.

**Comment by Aias00** [comment]
@johnlanni
